# ‘Affordable Places to Raise a Family’ Analysis

## <i>Data Work</i>

In [1]:
import pandas as pd

### Population, Migration and Births

<p>Let's read in the population dataset from 2010 - 2016. As a plus it has births and migration data we can use in the analysis.</p>
<p>Data is courtesy of the [U.S. Census Bureau](https://www2.census.gov/programs-surveys/popest/datasets/2010-2016/metro/totals/cbsa-est2016-alldata.csv).</p>

In [2]:
df = pd.read_csv("/Users/alexmahadevan/Desktop/Projects/Raise a Family/cbsa-est2016-alldata.csv")

<p>If people are moving to a county or metropolitan area, that probably means it's a good place to live.</p>

<p>So we'll go ahead and create a new variable that takes into account the average annual net migration to a region called AVG_NETMIG. The higher the number, the greater average migration to the municipality.</p>
<p>But, wait. Won't way more people be moving to a huge city like Miami than a smaller one like Sarasota?</p>
<p>To control for population amount, let's find the net migration per capita by diving net migration in each year by the population estimate for that year.</p>

In [3]:
df['AVG_NETMIG'] = (df['NETMIG2011']/df['POPESTIMATE2011'] +
df['NETMIG2012']/df['POPESTIMATE2012'] + df['NETMIG2013']/df['POPESTIMATE2013'] +
df['NETMIG2014']/df['POPESTIMATE2014'] + df['NETMIG2015']/df['POPESTIMATE2015'] +
df['NETMIG2016']/df['POPESTIMATE2016'])/6

<p>But, we also want to take into account the overall change in population between 2010 and 2016.</p>

<p>So we will create a new variable called POP_CHANGE.</p>

In [4]:
df['POP_CHANGE'] = (df['POPESTIMATE2016'] - df['POPESTIMATE2010']) / df['POPESTIMATE2010']

<p>You probably want to raise a family where other people are raising a family right?</p>
<p>This statistics may not be the best (since sometimes high birth rates correlate with poverty), but let's create a new variable for births per capita for 2016.</p>
<p>We shall call it BIRTHS_PER_CAPITA.</p>

In [5]:
df['BIRTHS_PER_CAPITA'] = df['BIRTHS2016']/df['POPESTIMATE2016']

<p>Are there any other variables we can create that would be important to someone looking to raise a family?</p>

<p>Not from this dataset, but while we're thinking about it, let's drill it down to the 382 metropolitan statistical areas.</p>

<p>First, let's count the number of types in the variable LSAD to make sure we have 382.</p>

In [6]:
df.LSAD.value_counts()

County or equivalent             1825
Micropolitan Statistical Area     551
Metropolitan Statistical Area     382
Metropolitan Division              31
Name: LSAD, dtype: int64

<p>Success! Now, let's drill it down by creating a new dataframe called MSA_POP.</p>

In [7]:
MSA_POP = df[df.LSAD == 'Metropolitan Statistical Area']

<p><b>Let's just leave that there for now, and explore other variables we can add to this analysis.</b></p>

### Museums

<p>I found a data set that has all of the museums in cities, let's have a look at that.</p>

In [8]:
MUSEUM = pd.read_csv("/Users/alexmahadevan/Desktop/Projects/Raise a Family/Workbook1.csv")

In [9]:
MUSEUM.head()

ADCITY ADSTATE       Unnamed: 2
0        SITKA      AK        SITKA, AK
1  KING SALMON      AK  KING SALMON, AK
2    FAIRBANKS      AK    FAIRBANKS, AK
3       PALMER      AK       PALMER, AK
4       KODIAK      AK       KODIAK, AK

<p>As you can see, we need to concatenate the city and state variables since there are a ton of similarly named cities throughout U.S. states</p>

In [10]:
MUSEUM['CITY'] = MUSEUM['ADCITY'] + ", " + MUSEUM['ADSTATE']

In [11]:
MUSEUM.CITY.value_counts()

NEW YORK, NY            297
CHICAGO, IL             193
WASHINGTON, DC          188
PHILADELPHIA, PA        179
LOS ANGELES, CA         162
HOUSTON, TX             133
BALTIMORE, MD           120
SAN FRANCISCO, CA       119
SEATTLE, WA             103
AUSTIN, TX               96
PORTLAND, OR             96
SAN DIEGO, CA            96
DALLAS, TX               94
BOSTON, MA               93
DENVER, CO               89
ATLANTA, GA              87
PITTSBURGH, PA           84
BROOKLYN, NY             77
LOUISVILLE, KY           76
CINCINNATI, OH           76
NEW ORLEANS, LA          72
MILWAUKEE, WI            70
RICHMOND, VA             69
SAN ANTONIO, TX          69
ALBUQUERQUE, NM          68
TUCSON, AZ               68
HUGO, OK                 67
CLEVELAND, OH            65
HONOLULU, HI             64
MIAMI, FL                64
                       ... 
HOUSTON TX, TX            1
FOREST PARK, GA           1
READS LANDING, MN         1
FREEPORT, MI              1
MOULTRIE, GA        

<p>Now, let's create a new database with each city and the amount of museums in each.</p>

In [12]:
MUSEUM_CITIES = MUSEUM.CITY.value_counts()

In [13]:
MUSEUM_CITIES.head()

NEW YORK, NY        297
CHICAGO, IL         193
WASHINGTON, DC      188
PHILADELPHIA, PA    179
LOS ANGELES, CA     162
Name: CITY, dtype: int64

<p>Eventually, we should adjust this for population, but 12,022 is a lot of cities to do that with.</p>
<p>So let's leave that database there for now until we pare down our metro data, then we can match the cities with metros that are statistically significant with their museum score.</p>

<p><b>Let's find some more data to mess with</b></p>

### Home Prices
<p>This data is courtesy of the [National Association of Realtors](https://www.nar.realtor/topics/metropolitan-median-area-prices-and-affordability).

<p>The numbers are in thousands.</p>

In [14]:
HOME_PRICES = pd.read_csv("/Users/alexmahadevan/Desktop/Projects/Raise a Family/Homeprices.csv")

In [15]:
HOME_PRICES.head(1)

CBSA Metropolitan Area   2014   2015   2016   2017 Change
0  10180       Abilene, TX  136.2  151.7  157.5  158.8   2.2%

<p>Again, we have created the dataframe, so let's just leave it here for now and continue.</p>

### Population, Migration and Births (Counties)
<p>Here we're doing essentially what we did above with metropolitan statistical areas, but with counties, to give us another option for a starting point.</p>
<p>Data is also courtesy of the [U.S. Census Bureau](https://www2.census.gov/programs-surveys/popest/datasets/2010-2016/counties/totals/co-est2016-alldata.csv)

In [16]:
df_COUNTY = pd.read_csv("/Users/alexmahadevan/Desktop/Projects/Raise a Family/COUNTY_POPULATION.csv")

<p>Just as we did with metros, let's find the average net migration for the county dataframe.</p>

In [17]:
df_COUNTY['AVG_NETMIG'] = (df_COUNTY['RNETMIG2011'] + df_COUNTY['RNETMIG2012'] +
                           df_COUNTY['RNETMIG2013'] + df_COUNTY['RNETMIG2014'] +
                           df_COUNTY['RNETMIG2015'] + df_COUNTY['RNETMIG2016']) / 6

In [18]:
df_COUNTY.head(1)

SUMLEV  REGION  DIVISION  STATE  COUNTY   STNAME  CTYNAME      CODE  \
0      40       3         6      1       0  Alabama  Alabama  1Alabama   

   CENSUS2010POP  ESTIMATESBASE2010     ...      RDOMESTICMIG2014  \
0        4779736            4780131     ...              0.086842   

   RDOMESTICMIG2015  RDOMESTICMIG2016  RNETMIG2011  RNETMIG2012  RNETMIG2013  \
0         -0.642255         -0.177829     0.709411     1.062617     1.202226   

   RNETMIG2014  RNETMIG2015  RNETMIG2016  AVG_NETMIG  
0     1.065267      0.41167     0.797351    0.874757  

[1 rows x 118 columns]

<p>Now let's look at population change.</p>

In [19]:
df_COUNTY['POP_CHANGE'] = (df_COUNTY['POPESTIMATE2016'] - df_COUNTY['POPESTIMATE2010']) / df_COUNTY['POPESTIMATE2010']

<p>And again, despite some questions about births as an indicator, let's add births per capita to this whole mix.</p>

In [20]:
df_COUNTY['BIRTHS_PER_CAPITA'] = df_COUNTY['BIRTHS2016']/df_COUNTY['POPESTIMATE2016']

In [21]:
df_COUNTY.head(1)

SUMLEV  REGION  DIVISION  STATE  COUNTY   STNAME  CTYNAME      CODE  \
0      40       3         6      1       0  Alabama  Alabama  1Alabama   

   CENSUS2010POP  ESTIMATESBASE2010        ...          RDOMESTICMIG2016  \
0        4779736            4780131        ...                 -0.177829   

   RNETMIG2011  RNETMIG2012  RNETMIG2013  RNETMIG2014  RNETMIG2015  \
0     0.709411     1.062617     1.202226     1.065267      0.41167   

   RNETMIG2016  AVG_NETMIG  POP_CHANGE  BIRTHS_PER_CAPITA  
0     0.797351    0.874757    0.016259            0.01204  

[1 rows x 120 columns]

In [22]:
df_COUNTY.sort_values("BIRTHS_PER_CAPITA", ascending=False).head(1)

SUMLEV  REGION  DIVISION  STATE  COUNTY  STNAME       CTYNAME  \
935      50       2         4     20      61  Kansas  Geary County   

               CODE  CENSUS2010POP  ESTIMATESBASE2010        ...          \
935  20Geary County          34362              34362        ...           

     RDOMESTICMIG2016  RNETMIG2011  RNETMIG2012  RNETMIG2013  RNETMIG2014  \
935        -67.137955   -24.821138    50.185671   -52.458578   -29.819912   

     RNETMIG2015  RNETMIG2016  AVG_NETMIG  POP_CHANGE  BIRTHS_PER_CAPITA  
935   -15.448733   -62.177023  -22.423285    0.008559           0.030012  

[1 rows x 120 columns]

<p>Hmmmm. Some very interesting counties ranking high there!</p>
<p>Let's leave that, and look at some economic statistics</p>

### Local Economic Indicators
<p>Data is courtesy of the [The U.S. Census Bureau](https://www.census.gov/data/tables/2016/demo/income-poverty/p60-256.html).

In [23]:
UNEMPLOYMENT = pd.read_csv("/Users/alexmahadevan/Desktop/Projects/Raise a Family/LOCAL_ECONOMICS.csv")

In [24]:
UNEMPLOYMENT.head(1)

CTYNAME      CODE Area_name  Unemployment_rate_2016  \
0  Alabama  1Alabama   Alabama                     6.0   

  Median_Household_Income_2015  Med_HH_Income_Percent_of_State_Total_2015  
0                       44,833                                      100.0

<p>You'll probably notice a weird variable in there called CODE. (Actually, since you're smart enough to look at the methodology for this study you probably noticed back in the county population data.)</p>
<p>I had to use some trickery to create a variable we could use to merge df_COUNTY and EMPLOYMENT. I did that by concatenating the state code number and the county name. I did this because there are tons of counties with the same name.</p>

In [25]:
UNEMPLOYMENT.CODE.head(1)

0    1Alabama
Name: CODE, dtype: object

<p>Now we just have to merge the two data frames, so we can work from the same place.</p> 
<p>This should give us a good snapshot of each county's economy, which we can use as a starting point for our analysis</p>

In [26]:
MERGED = pd.merge(df_COUNTY, UNEMPLOYMENT, on='CODE')

In [27]:
MERGED.head()

SUMLEV  REGION  DIVISION  STATE  COUNTY   STNAME       CTYNAME_x  \
0      40       3         6      1       0  Alabama         Alabama   
1      50       3         6      1       1  Alabama  Autauga County   
2      50       3         6      1       3  Alabama  Baldwin County   
3      50       3         6      1       5  Alabama  Barbour County   
4      50       3         6      1       7  Alabama     Bibb County   

              CODE  CENSUS2010POP  ESTIMATESBASE2010  \
0         1Alabama        4779736            4780131   
1  1Autauga County          54571              54571   
2  1Baldwin County         182265             182265   
3  1Barbour County          27457              27457   
4     1Bibb County          22915              22919   

                     ...                      RNETMIG2015  RNETMIG2016  \
0                    ...                         0.411670     0.797351   
1                    ...                        -2.654256     4.255281   
2                    ...                        18.585398    20.807611   
3                    ...                       -16.819716    -9.686991   
4                    ...                         0.887469     2.300681   

   AVG_NETMIG  POP_CHANGE  BIRTHS_PER_CAPITA       CTYNAME_y  \
0    0.874757    0.016259           0.012040         Alabama   
1    0.026241    0.012312           0.011387  Autauga County   
2   19.461493    0.138451           0.010903  Baldwin County   
3   -7.845474   -0.050570           0.009744  Barbour County   
4   -1.863208   -0.009536           0.011748     Bibb County   

            Area_name  Unemployment_rate_2016  Median_Household_Income_2015  \
0             Alabama                     6.0                        44,833   
1  Autauga County, AL                     5.3                        56,580   
2  Baldwin County, AL                     5.4                        52,387   
3  Barbour County, AL                     8.6                        31,433   
4     Bibb County, AL                     6.6                        40,767   

   Med_HH_Income_Percent_of_State_Total_2015  
0                                      100.0  
1                                      126.2  
2                                      116.8  
3                                       70.1  
4                                       90.9  

[5 rows x 125 columns]

In [28]:
MERGED.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3193 entries, 0 to 3192
Columns: 125 entries, SUMLEV to Med_HH_Income_Percent_of_State_Total_2015
dtypes: float64(41), int64(78), object(6)
memory usage: 3.1+ MB


<p>As you can see, we end up with 3,193 entries, so we didn't lose any observations. Nice.</p>
<p>Still, you'll also notice I left states in there. I might do something with those later.</p>

### Cities (ALL OF THEM!!!)

<p>Now the problem we're going to deal with, is that counties are not cities.</p>
<p>So, I've assembled a list of (what I believe is) every city in the U.S. Now we have to figure out how to match the data.</p>
<p>Let's load this sucker in.</p>

In [29]:
MERGED.to_csv("/Users/alexmahadevan/Desktop/ttttt.csv")

In [30]:
cities = pd.read_csv("/Users/alexmahadevan/Desktop/Projects/ALL_US_CITIES/csv/us-cities.csv")

In [31]:
cities.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45619 entries, 0 to 45618
Data columns (total 16 columns):
id               45619 non-null int64
name             45619 non-null object
county           45619 non-null object
state_code       45619 non-null object
state            45619 non-null object
zip_codes        45619 non-null object
type             45619 non-null object
latitude         45619 non-null float64
longitude        45619 non-null float64
area_code        45619 non-null object
population       45619 non-null int64
households       45619 non-null int64
median_income    45619 non-null int64
land_area        45619 non-null int64
water_area       45619 non-null int64
time_zone        45619 non-null object
dtypes: float64(2), int64(6), object(8)
memory usage: 5.6+ MB


<p>This looks clean, and with 45,619, we probably have most of the cities in the U.S.</p>

In [32]:
cities.head(1)

id  name                      county state_code   state zip_codes  type  \
0   1  Adak  Aleutians West Census Area         AK  Alaska     99546  City   

   latitude  longitude area_code  population  households  median_income  \
0     51.88 -176.65806       907         329          36          78500   

   land_area  water_area     time_zone  
0   88011658    88530676  America/Adak

<p>First we have to create a variable that is common between the two lists. We'll now concatenate the county name and state to create such a variable.</p>

In [33]:
cities['MATCH'] = cities['county'] + ", " + cities['state']

In [34]:
cities.head(1)

id  name                      county state_code   state zip_codes  type  \
0   1  Adak  Aleutians West Census Area         AK  Alaska     99546  City   

   latitude  longitude area_code  population  households  median_income  \
0     51.88 -176.65806       907         329          36          78500   

   land_area  water_area     time_zone                               MATCH  
0   88011658    88530676  America/Adak  Aleutians West Census Area, Alaska

In [35]:
MERGED['MATCH'] = MERGED['CTYNAME_x'] + ", " + MERGED['STNAME']

In [36]:
MERGED.head(1)

SUMLEV  REGION  DIVISION  STATE  COUNTY   STNAME CTYNAME_x      CODE  \
0      40       3         6      1       0  Alabama   Alabama  1Alabama   

   CENSUS2010POP  ESTIMATESBASE2010        ...         RNETMIG2016  \
0        4779736            4780131        ...            0.797351   

   AVG_NETMIG  POP_CHANGE  BIRTHS_PER_CAPITA  CTYNAME_y  Area_name  \
0    0.874757    0.016259            0.01204    Alabama    Alabama   

   Unemployment_rate_2016  Median_Household_Income_2015  \
0                     6.0                        44,833   

   Med_HH_Income_Percent_of_State_Total_2015             MATCH  
0                                      100.0  Alabama, Alabama  

[1 rows x 126 columns]

In [37]:
df_location = pd.merge(cities, MERGED, on='MATCH')

In [38]:
df_location.head()

id          name                      county state_code   state zip_codes  \
0    1          Adak  Aleutians West Census Area         AK  Alaska     99546   
1   21          Atka  Aleutians West Census Area         AK  Alaska     99547   
2   24  Attu Station  Aleutians West Census Area         AK  Alaska     96950   
3  225      Nikolski  Aleutians West Census Area         AK  Alaska     99638   
4  304    St. George  Aleutians West Census Area         AK  Alaska     99591   

   type  latitude  longitude area_code  \
0  City  51.88000 -176.65806       907   
1  City  52.19611 -174.20056       907   
2   CDP  52.88025  173.25608       907   
3   CDP  52.93806 -168.86778       907   
4  City  56.60000 -169.54167       907   

                     ...                      RNETMIG2015  RNETMIG2016  \
0                    ...                        -9.375814   -21.429826   
1                    ...                        -9.375814   -21.429826   
2                    ...                        -9.375814   -21.429826   
3                    ...                        -9.375814   -21.429826   
4                    ...                        -9.375814   -21.429826   

   AVG_NETMIG  POP_CHANGE  BIRTHS_PER_CAPITA                   CTYNAME_y  \
0   -1.020986    0.018211           0.005313  Aleutians West Census Area   
1   -1.020986    0.018211           0.005313  Aleutians West Census Area   
2   -1.020986    0.018211           0.005313  Aleutians West Census Area   
3   -1.020986    0.018211           0.005313  Aleutians West Census Area   
4   -1.020986    0.018211           0.005313  Aleutians West Census Area   

                        Area_name  Unemployment_rate_2016  \
0  Aleutians West Census Area, AK                     3.3   
1  Aleutians West Census Area, AK                     3.3   
2  Aleutians West Census Area, AK                     3.3   
3  Aleutians West Census Area, AK                     3.3   
4  Aleutians West Census Area, AK                     3.3   

   Median_Household_Income_2015  Med_HH_Income_Percent_of_State_Total_2015  
0                        80,695                                      110.0  
1                        80,695                                      110.0  
2                        80,695                                      110.0  
3                        80,695                                      110.0  
4                        80,695                                      110.0  

[5 rows x 142 columns]

In [39]:
df_location.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 44343 entries, 0 to 44342
Columns: 142 entries, id to Med_HH_Income_Percent_of_State_Total_2015
dtypes: float64(43), int64(84), object(15)
memory usage: 48.4+ MB


<p>Now, I'm going to send it off-site to examine and clean up in Excel.</p>

In [40]:
df_location.to_csv("/Users/alexmahadevan/Desktop/Projects/Raise a Family/locations.csv")

<p>OK, now I'll read it back in. I'm just going to use the standard df marking for our big data set.</p>

In [41]:
df = pd.read_csv("/Users/alexmahadevan/Desktop/Projects/Raise a Family/locations.csv")

In [42]:
df.head(1)

Unnamed: 0  id  name                      county state_code   state  \
0           0   1  Adak  Aleutians West Census Area         AK  Alaska   

  zip_codes  type  latitude  longitude  \
0     99546  City     51.88 -176.65806   

                     ...                     RNETMIG2015  RNETMIG2016  \
0                    ...                       -9.375814   -21.429826   

   AVG_NETMIG  POP_CHANGE  BIRTHS_PER_CAPITA                   CTYNAME_y  \
0   -1.020986    0.018211           0.005313  Aleutians West Census Area   

                        Area_name Unemployment_rate_2016  \
0  Aleutians West Census Area, AK                    3.3   

   Median_Household_Income_2015  Med_HH_Income_Percent_of_State_Total_2015  
0                        80,695                                      110.0  

[1 rows x 143 columns]

In [43]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 44343 entries, 0 to 44342
Columns: 143 entries, Unnamed: 0 to Med_HH_Income_Percent_of_State_Total_2015
dtypes: float64(43), int64(85), object(15)
memory usage: 48.4+ MB


In [44]:
df.sort_values('median_income' , ascending=True)

Unnamed: 0     id                             name  \
2504         2504   3252                          Storrie   
27008       27008  27918                          Winston   
3166         3166   2700                    Little Valley   
26995       26995  27885                     Trout Valley   
26987       26987  27665             Lake Roberts Heights   
3154         3154   2418                        Fort Dick   
26975       26975  27804                           Rivers   
26967       26967  27698                     Lower Frisco   
26955       26955  27743                          Newkirk   
26932       26932  27831                         San Luis   
235           235     93                 Eureka Roadhouse   
26815       26815  27725                          Moquino   
3082         3082   3077                    Robinson Mill   
26809       26809  27612                       Fence Lake   
26774       26774  27529                           Canova   
3065         3065   2203                    Clipper Mills   
26758       26758  27662                          La Joya   
26268       26268  27302                      Ross Corner   
296           296    327                          Tolsona   
39675       39675  40244                           Lopeño   
289           289    208                          Nabesna   
23558       23558  25000                      Rock Island   
23770       23770  24849                            Orrin   
277           277     53                          Chisana   
38998       38998  39514                         Brundage   
24346       24346  24560                           Larson   
2971         2971   3338  University of California-Merced   
25537       25537  26348                       White Clay   
3017         3017   3410                         Westside   
43990       43990  45059                           Bowden   
...           ...    ...                              ...   
28443       28443  29748                 Scarsdale (Town)   
23446       23446  24965                              Ree   
3447         3447   3556                          Crisman   
23784       23784  23828                         Big Bend   
23824       23824  25027                  Ross (Township)   
37321       37321  37931                             Eden   
23375       23375  24200                           Felson   
23888       23888  24914                    Pleasant Hill   
23787       23787  23986                       Clearwater   
23560       23560  25058                      Sauk Valley   
13167       13167  13532                       Washington   
23631       23631  23756              Arnegard (Township)   
24181       24181  24939                            Pratt   
36821       36821  37667                           Bangor   
40244       40244  41633                   Greenbackville   
37049       37049  38299                        Lone Rock   
36671       36671  38527                      Prairiewood   
23192       23192  23879                        Briarwood   
12051       12051  12551                          Jackson   
23215       23215  24282                             Gill   
23740       23740  24242                        Frederick   
24649       24649  25191                         Superior   
37221       37221  38192                        Jefferson   
37231       37231  38524                   Prairie Center   
23714       23714  25291                        Wadsworth   
24168       24168  24532                      Lake Hester   
23715       23715  25307                          Walters   
21211       21211  21731                       Moss Creek   
24861       24861  25605                           Corner   
23790       23790  24032                         Crowfoot   

                           county state_code          state  \
2504                Plumas County         CA     California   
27008               Sierra County         NM     New Mexico   
3166                Lassen County  

In [45]:
df['migration_z'] = (df['AVG_NETMIG'] - df.AVG_NETMIG.mean()) / df.AVG_NETMIG.std()

In [46]:
df.head(1)

Unnamed: 0  id  name                      county state_code   state  \
0           0   1  Adak  Aleutians West Census Area         AK  Alaska   

  zip_codes  type  latitude  longitude     ...      RNETMIG2016  AVG_NETMIG  \
0     99546  City     51.88 -176.65806     ...       -21.429826   -1.020986   

   POP_CHANGE  BIRTHS_PER_CAPITA                   CTYNAME_y  \
0    0.018211           0.005313  Aleutians West Census Area   

                        Area_name Unemployment_rate_2016  \
0  Aleutians West Census Area, AK                    3.3   

  Median_Household_Income_2015  Med_HH_Income_Percent_of_State_Total_2015  \
0                       80,695                                      110.0   

   migration_z  
0    -0.079202  

[1 rows x 144 columns]

In [47]:
df.head(20)

Unnamed: 0   id            name                      county state_code  \
0            0    1            Adak  Aleutians West Census Area         AK   
1            1   21            Atka  Aleutians West Census Area         AK   
2            2   24    Attu Station  Aleutians West Census Area         AK   
3            3  225        Nikolski  Aleutians West Census Area         AK   
4            4  304      St. George  Aleutians West Census Area         AK   
5            5  308        St. Paul  Aleutians West Census Area         AK   
6            6  338        Unalaska  Aleutians West Census Area         AK   
7            7    2          Akhiok       Kodiak Island Borough         AK   
8            8   10         Aleneva       Kodiak Island Borough         AK   
9            9   52         Chiniak       Kodiak Island Borough         AK   
10          10  150          Karluk       Kodiak Island Borough         AK   
11          11  167          Kodiak       Kodiak Island Borough         AK   
12          12  168  Kodiak Station       Kodiak Island Borough         AK   
13          13  182      Larsen Bay       Kodiak Island Borough         AK   
14          14  239      Old Harbor       Kodiak Island Borough         AK   
15          15  241        Ouzinkie       Kodiak Island Borough         AK   
16          16  266      Port Lions       Kodiak Island Borough         AK   
17          17  353      Womens Bay       Kodiak Island Borough         AK   
18          18    3        Akiachak          Bethel Census Area         AK   
19          19    4           Akiak          Bethel Census Area         AK   

     state      zip_codes  type  latitude  longitude     ...      RNETMIG2016  \
0   Alaska          99546  City  51.88000 -176.65806     ...       -21.429826   
1   Alaska          99547  City  52.19611 -174.20056     ...       -21.429826   
2   Alaska          96950   CDP  52.88025  173.25608     ...       -21.429826   
3   Alaska          99638   CDP  52.93806 -168.86778     ...       -21.429826   
4   Alaska          99591  City  56.60000 -169.54167     ...       -21.429826   
5   Alaska          99660  City  57.12222 -170.27500     ...       -21.429826   
6   Alaska  99685 / 99692  City  53.87361 -166.53667     ...       -21.429826   
7   Alaska          99615  City  56.94556 -154.17028     ...       -16.641837   
8   Alaska          99615   CDP  58.06000 -152.90972     ...       -16.641837   
9   Alaska          99615   CDP  57.63194 -152.18250     ...       -16.641837   
10  Alaska  99608 / 99615   CDP  57.57194 -154.45556     ...       -16.641837   
11  Alaska  99615 / 99619  City  57.79000 -152.40722     ...       -16.641837   
12  Alaska          99615   CDP  57.76795 -152.59591     ...       -16.641837   
13  Alaska  99615 / 99624  City  57.54000 -153.97861     ...       -16.641837   
14  Alaska          99643  City  57.20278 -153.30389     ...       -16.641837   
15  Alaska          99644  City  57.92361 -152.50222     ...       -16.641837   
16  Alaska          99550  City  57.86750 -152.88222     ...       -16.641837   
17  Alaska          99615   CDP  57.68222 -152.66861     ...       -16.641837   
18  Alaska          99551   CDP  60.90944 -161.43139     ...       -14.815640   
19  Alaska          99552  City  60.91222 -161.21389     ...       -14.815640   

    AVG_NETMIG  POP_CHANGE  BIRTHS_PER_CAPITA                   CTYNAME_y  \
0    -1.020986    0.018211           0.005313  Aleutians West Census Area   
1    -1.020986    0.018211           0.005313  Aleutians West Census Area   
2    -1.020986    0.018211           0.005313  Aleutians West Census Area   
3    -1.020986    0.018211           0.005313  Aleutians West Census Area   
4    -1.020986    0.018211           0.005313  Aleutians West Census Area   
5    -1.020986    0.018211           0.005313  Aleutians West Census Area   
6    -1.020986    0.018211           0.005313  Aleutians West Census Area   
7   -10.865037    0.005860           0.016239  

### Affordability data